geode-core/src/main/java/org/apache/geode/management/DiskStoreMXBean.java

```diff
@@ -196,6 +196,11 @@
    */
   float getDiskUsagePercentage();
 
+  /**
+   * Returns the free percentage of the disk storage
+   */
+  float getDiskFreePercentage();
```

Comment:
This is a public interface, even though I rarely see people manually implementing it, the addition of a new method would break backward compatibility... would it make sense to add a `default` implementation?.

Reply:
This PR is based on GEODE-5222, and in that PR it was considered ok to modify the interface to add "getDiskUsagePercentage" without default implementation, I suppose it was due to its not usually implemented as you said.

Reply:
I'm not sure why we didn't consider "worth it" adding a `default` implementation for `getDiskUsagePercentage` in [GEODE-5222](https://issues.apache.org/jira/browse/GEODE-5222), I'm tagging the original reviewers as they might have more context than me on this one.
Either way, unless there's a planned major version release in the near future, I just don't like the idea of breaking the backward compatibility in our public classes... even if they are rarely implemented by our users, we can't tell for sure.

@moleske @kirklund @aaronlindsey

Reply:
@jujoramos Does the addition of a new method on an interface break old code that is already compiled against the old interface? When you say it "breaks backward compatibility" what do you mean? Is it that the old class files will no longer run with the new product jar? Or is it that old code will no longer compile? In the past we treated method additions as compatible, but removing methods and changing existing methods as incompatible.

Reply:
@jujoramos I like the idea of adding a default to interface methods we add to existing external interfaces.

Reply:
@dschneider-pivotal: I'm not sure about what happens when code compiled against the old interface is executed using the new one, haven't tried yet.
That said, user code implementing the old version of the interface will certainly not compile if we don't provide a `default` implementation, that already sounds like an API change that is no backward compatible to me (please correct me if I'm wrong here).

Reply:
I have added default implementation for both methods. I suppose the default implementation of `getDiskUsagePercentage` should be included in 1.10 before release, right?

Reply:
Geode exposes much of the external apis on interfaces. Many of those interfaces are not meant to be implemented by geode users. Instead they are used to hide the internal geode implementation. For example Region is an interface that a geode user never needs to implement. So in the past as we added new features to geode adding a method to one of these interfaces was considered not breaking backward compatibility. But in the past we didn't even have the option of "default" methods on interfaces. The external geode api also has interfaces that only exist for geode users to implement. An example is CacheListener. It seems to me like a good rule going forward that if we add a method on an existing external interface that we use "default" on it to be safe. It can be unclear if an interface is meant to be implemented, and it is always on option and a geode user might implement Region for testing purposes or to have their own wrapper around the actual geode Region.

Reply:
> I'm not sure why we didn't consider "worth it" adding a `default` implementation for `getDiskUsagePercentage` in [GEODE-5222](https://issues.apache.org/jira/browse/GEODE-5222), I'm tagging the original reviewers as they might have more context than me on this one.

I think it's a good idea to add a default implementation for both methods. At the time I was very new to this project so I didn't consider the possibility of a breaking change. I'm not sure whether this would be considered a "critical" fix to make it in 1.10. I suppose this might be a good question for the dev list.